core/src/main/java/com/datastrato/gravitino/catalog/rel/BaseTable.java

```diff
@@ -37,6 +38,29 @@ public abstract class BaseTable implements Table {
 
   @Nullable protected Index[] indexes;
 
+  private volatile TableOperations ops;
+
+  /** @return The {@link TableOperations} instance associated with this table. */
+  protected abstract TableOperations newOps();
+
+  /**
+   * Retrieves the {@link TableOperations} instance associated with this table. If the instance is
+   * not initialized, it is initialized and returned.
+   *
+   * @return The {@link TableOperations} instance associated with this table.
+   */
+  public TableOperations ops() {
```

Comment:
For JDBC/Iceberg table, this will always return null and throw NPE potentially. You should not return null, instead a specified exception is required.

Reply:
fixed